src/vstest.console/TestPlatformHelpers/TestRequestManager.cs

```diff
@@ -249,6 +275,10 @@ public bool RunTests(TestRunRequestPayload testRunRequestPayload, ITestHostLaunc
             var success = this.RunTests(requestData, runCriteria, testRunEventsRegistrar, protocolConfig);
             EqtTrace.Info("TestRequestManager.RunTests: run tests completed, sucessful: {0}.", success);
             this.testPlatformEventSource.ExecutionRequestStop();
+
+            this.metricsPublisher.PublishMetrics(TelemetryDataConstants.TestExecutionCompleteEvent, requestData.MetricsCollection.Metrics);
+            requestData.MetricsCollection.Clear();
```

Comment:
I don't think it is required, as we are always creating a new MetricsCollection for Discovery and Execution flows.